[angleproject] linking fails in mingw32 with undefined symbols

### Description / Steps to reproduce the issue

When attempting to link an sdl2/gles2 hello world against ANGLE's GLESv2, I get a bunch of undefined references.

```
$ gcc sdl2gles2tri.c -lmingw32 -mwindows -lSDL2main -lSDL2 -lGLESv2
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0x49): undefined reference to `_imp__glCreateShader@4'
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0x83): undefined reference to `_imp__glShaderSource@16'
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0x93): undefined reference to `_imp__glCompileShader@4'
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0xb2): undefined reference to `_imp__glGetShaderiv@12'
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0xe3): undefined reference to `_imp__glGetShaderiv@12'
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0x11f): undefined reference to `_imp__glGetShaderInfoLog@16'
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0x14d): undefined reference to `_imp__glDeleteShader@4'
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0x236): undefined reference to `_imp__glCreateProgram@0'
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0x25d): undefined reference to `_imp__glAttachShader@8'
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0x274): undefined reference to `_imp__glAttachShader@8'
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0x294): undefined reference to `_imp__glBindAttribLocation@12'
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0x2a4): undefined reference to `_imp__glLinkProgram@4'
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0x2c6): undefined reference to `_imp__glGetProgramiv@12'
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0x300): undefined reference to `_imp__glGetProgramiv@12'
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0x345): undefined reference to `_imp__glGetProgramInfoLog@16'
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0x373): undefined reference to `_imp__glDeleteProgram@4'
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0x39b): undefined reference to `_imp__glClearColor@16'
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0x416): undefined reference to `_imp__glViewport@16'
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0x427): undefined reference to `_imp__glClear@4'
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0x437): undefined reference to `_imp__glUseProgram@4'
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0x46f): undefined reference to `_imp__glVertexAttribPointer@24'
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0x480): undefined reference to `_imp__glEnableVertexAttribArray@4'
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0x4a1): undefined reference to `_imp__glDrawArrays@12'
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0x5c6): undefined reference to `_imp__glGetError@0'
C:/msys64/mingw32/bin/sdl2gles2tri.c:(.text+0x5e5): undefined reference to `_imp__glGetError@0'
```

### Expected behavior

Proper linkage.

### Actual behavior

As mentioned, failure to resolve ABI.

Other relevant information,  ANGLE's libraries don't seem to export these symbols.
```
$ nm -g libGLESv2.dll.a | grep _imp__glCreateShader@4
(no output)
```

```
$ nm -g libGLESv2.dll.a | grep glCreateShader
00000000 I __imp__glCreateShaderProgramvEXT
00000000 T _glCreateShaderProgramvEXT
00000000 I __imp__glCreateShaderProgramv
00000000 T _glCreateShaderProgramv
00000000 I __imp__glCreateShader
00000000 T _glCreateShader
```

I'm not sure where the source of this `@4` is being introduced, but the number of underscores at the beginning of the function signature seems incorrect too. I've only tried on MINGW32, as SDL seems to require linking against specifically the 32bit mingw library.

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MINGW32_NT-10.0-22621

### MINGW environments affected

- [ ] MINGW64
- [X] MINGW32
- [ ] UCRT64
- [ ] CLANG64
- [ ] CLANG32
- [ ] CLANGARM64

### Are you willing to submit a PR?

If I can determine the solution, absolutely

--- a/mingw-w64-angleproject/003-angle-src-fixes.patch
+++ b/mingw-w64-angleproject/003-angle-src-fixes.patch
@@ -1,3 +1,16 @@
+diff --git a/src/compiler/translator/Compiler.cpp b/src/compiler/translator/Compiler.cpp
+index 37ca80a28..93b542bbb 100644
+--- a/src/compiler/translator/Compiler.cpp
++++ b/src/compiler/translator/Compiler.cpp
+@@ -1217,7 +1217,7 @@ bool TCompiler::checkAndSimplifyAST(TIntermBlock *root,
+
+ bool TCompiler::resizeClipAndCullDistanceBuiltins(TIntermBlock *root)
+ {
+-    auto resizeVariable = [=](const ImmutableString &name, uint32_t size, uint32_t maxSize) {
++    auto resizeVariable = [=, this](const ImmutableString &name, uint32_t size, uint32_t maxSize) {
+         // Skip if the variable is not used or implicitly has the maximum size
+         if (size == 0 || size == maxSize)
+             return true;
 diff --git a/src/common/mathutil.h b/src/common/mathutil.h
 index d142b35de..46b9eac13 100644
 --- a/src/common/mathutil.h

--- a/mingw-w64-angleproject/PKGBUILD
+++ b/mingw-w64-angleproject/PKGBUILD
@@ -6,7 +6,7 @@ _realname=angleproject
 pkgbase=mingw-w64-${_realname}
 pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
 pkgver=2.1.r20724.b26b01a2
-pkgrel=2
+pkgrel=3
 pkgdesc='A conformant OpenGL ES implementation for Windows, Mac, Linux, iOS and Android (mingw-w64)'
 arch=('any')
 mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32' 'clangarm64')
@@ -45,7 +45,7 @@ sha256sums=('SKIP'
             'SKIP'
             '51daba3ef168035b0ed89c0f56a4ee24c1b4c9d234ec126c555fae689b22880e'
             '36e0d51262c28d59f78b81a05499de502cf8ab54fa5e2a3a3bcfb95c70337f8c'
-            '72b7287ca0d2e6b617e3251fbe9c5d07c32e7c0135b0f2685d4c970acd1ecbf0'
+            '6eb17823650278fd018ea82969d9f2c502d18c40b247607c62048aeab3a2d2c2'
             'bf878de3203a62e2a7f81bd4cd79237adc74804b4b692d9277da005ed56d4f2a'
             '86013781c2700219d4f64d7ac34ad16c40fcca9a641371385f67f642e23c643b'
             'f0fb05348bf2de599eff35e2d35e3336b9720b6bd2799af9d186c05ff45d34f0'

